### PR TITLE
[16.0][ADD] purchase_reorder_control

### PR DESCRIPTION
**Pull request by trisdoan**

### CODE REVIEW

File: purchase_reorder_control/tests/test_unpurchaseable_product.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+from odoo.exceptions import UserError
+from odoo.fields import Command
+
+from odoo.addons.product.tests.common import ProductAttributesCommon
+
+
+class TestUnpurchaseableProduct(ProductAttributesCommon):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
```
Comment (santostelmo) [inline purchase_reorder_control/tests/test_unpurchaseable_product.py:13]:
```suggestion
        super().setUpClass()
        cls.env = cls.env["base"].with_context(**DISABLED_MAIL_CONTEXT).env
```

**Review by santostelmo — APPROVED** [review]

File: purchase_reorder_control/tests/test_unpurchaseable_product.py (hunk 1 above)
Comment (nilshamerlinck) [reply purchase_reorder_control/tests/test_unpurchaseable_product.py:13]: I believe this is already done in odoo's [BaseCommon](https://github.com/odoo/odoo/blob/16.0/odoo/addons/base/tests/common.py#L31) and we have `ProductAttributesCommon < ProductCommon < BaseCommon`

Comment (santostelmo) [reply purchase_reorder_control/tests/test_unpurchaseable_product.py:13]: indeed it is done already in `BaseCommon`

**Review by cyrilmanuel — APPROVED** [review]

**Comment by HviorForgeFlow** [comment]
Just a comment. Have you considr archieving the Orderpoint instead of unlinking? Maybe it can become active again when purchase_ok become True again (or not), to avoid possible forgotten orderpoints to be active.

**Comment by trisdoan** [comment]
> Just a comment. Have you considr archieving the Orderpoint instead of unlinking? Maybe it can become active again when purchase_ok become True again (or not), to avoid possible forgotten orderpoints to be active.

Hi @HviorForgeFlow, it's a good idea.

I updated as you suggest :pray:

**Review by HviorForgeFlow — APPROVED** [review]

**Pull request merged**